### ISSUE TITLE
Hot reload error 108: Isolate is reloading

### ISSUE THREAD

**Issue by mit-mit** [flutter/flutter#14517]
Encountered this while hot reloading:

```
Error 108 received from application: Isolate is reloading
Unhandled exception:
NoSuchMethodError: The method '[]' was called on null.
Receiver: null
Tried calling: []("type")
#0      Object.noSuchMethod (dart:core-patch/dart:core/object_patch.dart:46)
#1      HotRunner.validateReloadReport (package:flutter_tools/src/run_hot.dart:415)
#2      HotRunner._reloadSources.<anonymous closure> (package:flutter_tools/src/run_hot.dart:542)
#3      _rootRunUnary (dart:async/zone.dart:1134)
#4      _CustomZone.runUnary (dart:async/zone.dart:1031)
#5      _FutureListener.handleValue (dart:async/future_impl.dart:129)
#6      _Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:633)
#7      _Future._propagateToListeners (dart:async/future_impl.dart:662)
#8      _Future._completeWithValue (dart:async/future_impl.dart:477)
#9      Future.wait.<anonymous closure> (dart:async/future.dart:392)
#10     _rootRunUnary (dart:async/zone.dart:1134)
#11     _CustomZone.runUnary (dart:async/zone.dart:1031)
#12     _FutureListener.handleValue (dart:async/future_impl.dart:129)
#13     _Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:633)
#14     _Future._propagateToListeners (dart:async/future_impl.dart:662)
#15     _Future._complete (dart:async/future_impl.dart:467)
#16     _SyncCompleter.complete (dart:async/future_impl.dart:51)
#17     _completeOnAsyncReturn (dart:async-patch/dart:async/async_patch.dart:255)
#18     Isolate.reloadSources (package:flutter_tools/src/vmservice.dart:1047)
#19     _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/dart:async/async_patch.dart:40)
#20     _rootRunUnary (dart:async/zone.dart:1134)
#21     _CustomZone.runUnary (dart:async/zone.dart:1031)
#22     _FutureListener.handleValue (dart:async/future_impl.dart:129)
#23     _Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:633)
#24     _Future._propagateToListeners (dart:async/future_impl.dart:662)
#25     _Future._complete (dart:async/future_impl.dart:467)
#26     _SyncCompleter.complete (dart:async/future_impl.dart:51)
#27     _completeOnAsyncReturn (dart:async-patch/dart:async/async_patch.dart:255)
#28     VM.invokeRpcRaw (package:flutter_tools/src/vmservice.dart:748)
#29     _asyncErrorWrapperHelper.<anonymous closure> (dart:async-patch/dart:async/async_patch.dart:47)
#30     _rootRunBinary (dart:async/zone.dart:1146)
#31     _CustomZone.runBinary (dart:async/zone.dart:1039)
#32     _FutureListener.handleError (dart:async/future_impl.dart:141)
#33     _Future._propagateToListeners.handleError (dart:async/future_impl.dart:645)
#34     _Future._propagateToListeners (dart:async/future_impl.dart:666)
#35     _Future._completeError (dart:async/future_impl.dart:485)
#36     _Future.timeout.<anonymous closure> (dart:async/future_impl.dart:736)
#37     _rootRunBinary (dart:async/zone.dart:1146)
#38     _CustomZone.runBinary (dart:async/zone.dart:1039)
#39     _FutureListener.handleError (dart:async/future_impl.dart:141)
#40     _Future._propagateToListeners.handleError (dart:async/future_impl.dart:645)
#41     _Future._propagateToListeners (dart:async/future_impl.dart:666)
#42     _Future._completeError (dart:async/future_impl.dart:485)
#43     _SyncCompleter._completeError (dart:async/future_impl.dart:55)
#44     _Completer.completeError (dart:async/future_impl.dart:27)
#45     Future.any.<anonymous closure> (dart:async/future.dart:454)
#46     _rootRunBinary (dart:async/zone.dart:1146)
#47     _CustomZone.runBinary (dart:async/zone.dart:1039)
#48     _FutureListener.handleError (dart:async/future_impl.dart:141)
#49     _Future._propagateToListeners.handleError (dart:async/future_impl.dart:645)
#50     _Future._propagateToListeners (dart:async/future_impl.dart:666)
#51     _Future._completeError (dart:async/future_impl.dart:485)
#52     _SyncCompleter._completeError (dart:async/future_impl.dart:55)
#53     _Completer.completeError (dart:async/future_impl.dart:27)
#54     Client._handleSingleResponse (package:json_rpc_2/src/client.dart:191)
#55     Client._handleResponse (package:json_rpc_2/src/client.dart:179)
#56     _rootRunUnary (dart:async/zone.dart:1134)
#57     _CustomZone.runUnary (dart:async/zone.dart:1031)
#58     _CustomZone.runUnaryGuarded (dart:async/zone.dart:933)
#59     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:330)
#60     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:257)
#61     _StreamController&&_SyncStreamControllerDispatch._sendData (dart:async/stream_controller.dart:762)
#62     _StreamController._add (dart:async/stream_controller.dart:638)
#63     _StreamController.add (dart:async/stream_controller.dart:584)
#64     Peer.listen.<anonymous closure> (package:json_rpc_2/src/peer.dart:91)
#65     _rootRunUnary (dart:async/zone.dart:1134)
#66     _CustomZone.runUnary (dart:async/zone.dart:1031)
#67     _CustomZone.runUnaryGuarded (dart:async/zone.dart:933)
#68     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:330)
#69     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:257)
#70     _ForwardingStreamSubscription._add (dart:async/stream_pipe.dart:132)
#71     _MapStream._handleData (dart:async/stream_pipe.dart:232)
#72     _ForwardingStreamSubscription._handleData (dart:async/stream_pipe.dart:164)
#73     _rootRunUnary (dart:async/zone.dart:1134)
#74     _CustomZone.runUnary (dart:async/zone.dart:1031)
#75     _CustomZone.runUnaryGuarded (dart:async/zone.dart:933)
#76     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:330)
#77     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:257)
#78     _ForwardingStreamSubscription._add (dart:async/stream_pipe.dart:132)
#79     _ForwardingStream._handleData (dart:async/stream_pipe.dart:98)
#80     _ForwardingStreamSubscription._handleData (dart:async/stream_pipe.dart:164)
#81     _rootRunUnary (dart:async/zone.dart:1134)
#82     _CustomZone.runUnary (dart:async/zone.dart:1031)
#83     _CustomZone.runUnaryGuarded (dart:async/zone.dart:933)
#84     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:330)
#85     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:257)
#86     _StreamController&&_SyncStreamControllerDispatch._sendData (dart:async/stream_controller.dart:762)
#87     _StreamController._add (dart:async/stream_controller.dart:638)
#88     _rootRunUnary (dart:async/zone.dart:1134)
#89     _CustomZone.runUnary (dart:async/zone.dart:1031)
#90     _CustomZone.runUnaryGuarded (dart:async/zone.dart:933)
#91     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:330)
#92     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:257)
#93     _StreamController&&_SyncStreamControllerDispatch._sendData (dart:async/stream_controller.dart:762)
#94     _StreamController._add (dart:async/stream_controller.dart:638)
#95     _StreamController.add (dart:async/stream_controller.dart:584)
#96     new _WebSocketImpl._fromSocket.<anonymous closure> (dart:_http/websocket_impl.dart:1128)
#97     _rootRunUnary (dart:async/zone.dart:1134)
#98     _CustomZone.runUnary (dart:async/zone.dart:1031)
#99     _CustomZone.runUnaryGuarded (dart:async/zone.dart:933)
#100    _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:330)
#101    _BufferingStreamSubscription._add (dart:async/stream_impl.dart:257)
#102    _SinkTransformerStreamSubscription._add (dart:async/stream_transformers.dart:68)
#103    _EventSinkWrapper.add (dart:async/stream_transformers.dart:15)
#104    _WebSocketProtocolTransformer._messageFrameEnd (dart:_http/websocket_impl.dart:336)
#105    _WebSocketProtocolTransformer.add (dart:_http/websocket_impl.dart:231)
#106    _SinkTransformerStreamSubscription._handleData (dart:async/stream_transformers.dart:120)
```

**Comment by DanTup** [comment]
What version of Dart Code was this with? I think it's https://github.com/Dart-Code/Dart-Code/issues/492 but in v1.8.0 we should generally prevent sending concurrent reloads.

If the concurrent reload does get to flutter, the crash is #14184 which is fixed in #14494 (which shipped in flutter last night).

Let me know if you can still repro, and if so what you're doing (combination of saving multiple files and/or clicking the debugger restart icon while a reload is in progress where the previous causes).

**Comment by DanTup** [comment]
@mit-mit Have you seen this recently? I think probably it's the issues mentioned above and is fixed?
